tests/regression/test_interpolate.py

```diff
@@ -358,3 +358,108 @@ def test_interpolate_periodic_coords_max():
     assert np.allclose(np.unique(continuous.dat.data_ro),
                        [0.25, 0.5, 0.75, 1])
+
+
+# New FIAT Element
+
+from FIAT.dual_set import DualSet
+from FIAT.finite_element import CiarletElement
+from FIAT.reference_element import UFCInterval
+from FIAT.functional import PointEvaluation, IntegralMoment
+from FIAT.quadrature import make_quadrature
+from FIAT.polynomial_set import ONPolynomialSet
+ufcint = UFCInterval()
+
+
+class FIAT_C0ModifiedDualSet(DualSet):
+
+    def __init__(self, cell, order):
+        assert cell == ufcint and order == 3
+        entity_ids = {0: {0: [0], 1: [1]},
+                      1: {0: [2, 3]}}
+        vertnodes = [PointEvaluation(ufcint, xx)
+                     for xx in ufcint.vertices]
+        Q = make_quadrature(ufcint, 3)
+        # 1st integral moment node is integral(1*f(x)*dx)
+        ones = np.asarray([1.0 for x in Q.pts])
+        # 2nd integral moment node is integral(x*f(x)*dx)
+        xs = np.asarray([x for (x,) in Q.pts])
+        intnodes = [IntegralMoment(ufcint, Q, ones),
+                    IntegralMoment(ufcint, Q, xs)]
+        nodes = vertnodes + intnodes
+        super(FIAT_C0ModifiedDualSet, self).__init__(nodes, ufcint, entity_ids)
+
+
+class FIAT_C0Modified(CiarletElement):
+    r"""
+    An element, only defined for order 3 on interval cells, with 4 nodes:
+    ..math::
+     phi_0'(f) = f(0.)
+     phi_1'(f) = f(1.)
+     pfi_2'(f) = \int_{0}^{1} f(x) \,dx
+     pfi_3'(f) = \int_{0}^{1} x f(x) \,dx
+     """
+
+    def __init__(self, cell, order):
+        assert cell == ufcint and order == 3
+        poly_set = ONPolynomialSet(ufcint, 3)
+        dual_set = FIAT_C0ModifiedDualSet(ufcint, 3)
+        super(FIAT_C0Modified, self).__init__(poly_set, dual_set, 3, 0)
+
+
+def test_fiat_c0modified():
+    # can create
+    el = FIAT_C0Modified(UFCInterval(), 3)
+    # dual eval gives expected values from sum of point evaluations
+    fns = (lambda x: x[0], lambda x: x[0]**2)
+    expected = ([0, 1, 1/2, 1/3], [0, 1, 1/3, 1/4])
+    for i in range(len(fns)):
+        node_vals = []
+        for node in el.dual.nodes:
+            pt_dict = node.pt_dict
+            node_val = 0.0
+            for pt in pt_dict:
+                for (w, _) in pt_dict[pt]:
+                    node_val += w * fns[i](pt)
+            node_vals.append(node_val)
+        assert len(node_vals) == len(expected[i])
+        assert np.allclose(node_vals, expected[i])
+
+
+# FInAT Equivalent
+from finat.fiat_elements import ScalarFiatElement
+
+
+class FInAT_C0Modified(ScalarFiatElement):
+    def __init__(self, cell, degree):
+        super(FInAT_C0Modified, self).__init__(FIAT_C0Modified(cell, degree))
+
+
+# Add to TSFC Supported Elements
+import tsfc
+tsfc.finatinterface.supported_elements["C0 Modified"] = FInAT_C0Modified
+
+
+# Register in UFL
+import ufl
+ufl.register_element("C0 Modified", "C0Mod", 0, H1, "identity", (3, 3), "interval")
```

Comment:
Need to replace with variant as suggested by @wence- but not sure of syntax yet

Reply:
`FiniteElement(interval, "CG", 3, variant="interior-moment"`)` ?

Reply:
I'm afraid I don't follow `ufl.FiniteElement("CG", "interval", 3, variant="interior-moment")`  just creates a new ufl `ufl.FiniteElement` object rather than registering a name string which I surely need in order for the element to be created using `firedrake.FunctionSpace`

Reply:
```python
e = ufl.FiniteElement("CG", "interval", 3, variant="interior-moment")
V = FunctionSpace(mesh, e)
```

So now you just need the finatinterface to know how to handle the "interior-moment" variant.

Reply:
Ah - but then I need to modify `tsfc`'s `tsfc/finatinterface.py::convert_finiteelement()` function (see below). There is no externally modifiable TSFC interface for doing this as far as I can tell so I would actually have to change TSFC to let me support this element. My current solution at least allows everything to be gathered into one place. This seems to be exactly the point @rckirby noted in [this comment](https://github.com/FInAT/FInAT/issues/75#issuecomment-781471185) in https://github.com/FInAT/FInAT/issues/75.
```python
# Base finite elements first
@convert.register(ufl.FiniteElement)
def convert_finiteelement(element, **kwargs):
    cell = as_fiat_cell(element.cell())

    ...
    lmbda = supported_elements[element.family()]
    ...

    kind = element.variant()
    if kind is None:
        kind = 'spectral' if element.cell().cellname() == 'interval' else 'equispaced'  # default variant

    if element.family() == "Lagrange":
        if kind == 'equispaced':
            lmbda = finat.Lagrange
        elif kind == 'spectral' and element.cell().cellname() == 'interval':
            lmbda = finat.GaussLobattoLegendre
        elif kind in ['mgd', 'feec', 'qb', 'mse']:
            degree = element.degree()
            shift_axes = kwargs["shift_axes"]
            restriction = kwargs["restriction"]
            deps = {"shift_axes", "restriction"}
            return finat.RuntimeTabulated(cell, degree, variant=kind, shift_axes=shift_axes, restriction=restriction), deps
        else:
            raise ValueError("Variant %r not supported on %s" % (kind, element.cell()))  # Fails here
```
Is there some ability for me to override this function that I am missing?

Reply:
You have to replace this dispatch with your own one.

```
from tsfc.finatinterface import convert

old_converter = convert.dispatch(ufl.FiniteElement)

def my_converter(element, **kwargs):
    if element.variant() == "interior-moment" and element.family() == "CG":
        return finat.FancyElement(...)
    else:
        return old_convert(element, **kwargs)

convert.register(ufl.FiniteElement, my_converter)

# To undo this afterr the test has finished do

convert.register(ufl.FiniteElement, old_convert)
```

Reply:
Yes this is ugly, we're exactly trying to see where it is ugly and what we would need to do to improve this.

Reply:
Done, see 91e98aa9e9db0142339013740851ae823f48620f